src/js/front-end/setup/general/accept-action.js

```diff
@@ -84,11 +85,24 @@ const actionToFunction = (action) => {
     };
 };
 
-export const acceptAction = (user, action, parameters, isStateImport = false) => {
+export const acceptAction = (user, action, parameters, isStateImport = false, isFromReplay = false) => {
+    if (replayBlock("action",action,isFromReplay)){
+        return;
+    }
     const emit = (user === 'self' || isStateImport) ? true : false;
     if (parameters){
-        actionToFunction(action)(user, ...parameters, emit);
+        if(['shuffleAll','shuffleBottom','discardAll','lostZoneAll','handAll','leaveAll','VSTARGXFunction'].includes(action)){
+            actionToFunction(action)(user, ...parameters, emit, isFromReplay);
+        }
+        else{
+            actionToFunction(action)(user, ...parameters, emit);
+        }
     } else {
-        actionToFunction(action)(user, emit);
+        if(['shuffleAll','shuffleBottom','discardAll','lostZoneAll','handAll','leaveAll','VSTARGXFunction'].includes(action)){
```

Comment:
i think you'll be able to remove this check once the buttons/keybinds for all of these functions are hidden/disabled.

Reply:
Done in [83be7ec](https://github.com/xxmichaellong/ptcg-sim/pull/19/commits/83be7eca0da088e1045be0ea3836abff653fd95b), although again I left the check for VSTAR and GX